dashboard/src/modules/components/DatePickerComponent/index.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from "react";
+import {
+  InputGroup,
+  InputGroupText,
+  DatePicker,
+  isValidDate,
+  yyyyMMddFormat,
+  Button,
+} from "@patternfly/react-core";
+import "./index.css"
+import { formatDate } from "../../../utils/dateFormatter";
+
+function DatePickerWidget({
+  dataArray,
+  setPublicData,
+  controllerName,
+  setDateRange,
+}) {
+  const [fromDate, setFromDate] = useState(
+    formatDate(new Date(1990,10,4))
+  );
+  const [toDate, setToDate] = useState(
+    formatDate(new Date())
+  );
+  const toValidator = (date) =>
+    isValidDate(fromDate) && date >= fromDate
+      ? ""
+      : "To date must be less than from date";
```

Comment:
This toValidator function is meant for only one thing , to display an error message if the todate is less than fromDate ,so yes I have to change this error message.

Regarding adding `isValidDate(date)` condition ,it wont make any difference ,since if the date will be invalid(like an ill formatted or out of range one), the datePicker will use Patternfly default validator and will throw a Invalid date error automatically,so we don't need to take care about that

Reply:
I'm not sure, but I _think_, given the way Patternfly seems to be approaching this, that the `toValidator()`, which checks only the "to" date, won't ever be called if the "from" date is invalid, because, in that case, the "to" picker will be disabled.

If that's correct, then I don't see any point in checking the "from" date here.  (In fact, if the check at line 26 were to fail because the "from" date is invalid, it would result in an inappropriate error message!)

The `validators` attribute takes a list of validator functions, but specifying it appears to completely replace the default value...which is an empty list.  However, [the `DatePicker` code](https://github.com/patternfly/patternfly-react/blob/cc80f358e790331d92df19b49df25488b095265b/packages/react-core/src/components/DatePicker/DatePicker.tsx#L62) checks the date (using `isValidDate()`) at several _other_ points, so I suspect that we don't actually need to check (either the "to" or the "from") here at all.  (But, I don't speak Javascript well enough to be sure.)